### PR TITLE
ci: add slack notification to GH actions

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
This is how it looked for test PR: 
image

https://github.com/cilium/cilium/pull/16219

**Review by aanm — CHANGES_REQUESTED** [review]
Why does it matter if the test was cancelled?

**Comment by nebril** [comment]
@aanm cancellation can be caused by a timeout, which will allow us to investigate whether the timeout should be increased, or something else causing timeouts started happening.

File: .github/workflows/aks.yaml [hunk 1]
```diff
@@ -230,3 +230,12 @@ jobs:
           description: Connectivity test cancelled
           state: pending
           target_url: ${{ env.check_url }}
+
+      - name: Send slack notification
+        if: ${{ (cancelled() || failure()) && (github.event_name == 'schedule' || github.event_name == 'push') }}
```
Comment (pchaigno) [inline .github/workflows/aks.yaml:235]: I think we need something more fine-grained for `cancelled()` here (or we need to remove it). Now that we merged https://github.com/cilium/cilium/pull/16199, all outdated GitHub jobs will be cancelled (for example when we push a new version of a PR's branch). That will generated a lot of Slack notifications.

File: .github/workflows/eks.yaml [hunk 2]
```diff
@@ -223,3 +223,12 @@ jobs:
           description: Connectivity test cancelled
           state: pending
           target_url: ${{ env.check_url }}
+
+      - name: Send slack notification
+        if: ${{ (cancelled() || failure()) && (github.event_name == 'schedule' || github.event_name == 'push') }}
```
Comment (pchaigno) [inline .github/workflows/eks.yaml:228]: When is `github.event_name == 'push'` true? I thought this workflow was only triggered on PR comments or and schedule.

**Review by pchaigno — CHANGES_REQUESTED** [review]
This is great :rocket: 

Besides the `cancelled()` issue mentioned below, I'm a bit concerned about the number of notifications we will receive though. Where are the notifications sent? Will we receive notifications for every single check failing, in every single pull request?

File: .github/workflows/eks.yaml (hunk 2 above)
Comment (nebril) [reply .github/workflows/eks.yaml:228]: I copy pasted the same step to all workflows regardless of whether the `push` was actually in the file or not.

File: .github/workflows/aks.yaml (hunk 1 above)
Comment (nebril) [reply .github/workflows/aks.yaml:235]: @pchaigno this bit `(github.event_name == 'schedule' || github.event_name == 'push')` will only generate notifications for failing/cancelled builds that run on schedule, or on merge (`push` event). I did this so we are able to monitor closely what state the master branch (and possibly the stable branches) are in. No PR failure will cause a notification, this should be monitored by the PR author on his own.

Comment (pchaigno) [reply .github/workflows/aks.yaml:235]: Makes sense :+1:

File: .github/workflows/eks.yaml (hunk 2 above)
Comment (pchaigno) [reply .github/workflows/eks.yaml:228]: Ok, maybe it's easier/cleaner that way.

**Review by pchaigno — APPROVED** [review]

**Review by aanm — APPROVED** [review]

File: .github/workflows/aks.yaml (hunk 1 above)
Comment (christarazi) [inline .github/workflows/aks.yaml:235]: In cilium-cli, this was changed to `!success()`. Should we also make that consistent here as well? https://github.com/cilium/cilium-cli/pull/237#discussion_r644024948

Comment (pchaigno) [reply .github/workflows/aks.yaml:235]: :shipit:

**Pull request merged**